src/lib.rs

```diff
@@ -1,13 +1,446 @@
+        let cursor = if supports_ansi() {
+            Box::new(AnsiCursor::new()) as Box<(dyn Cursor + Sync + Send)>
+        } else {
+            Box::new(WinApiCursor::new()) as Box<(dyn Cursor + Sync + Send)>
+        };
+
+        #[cfg(unix)]
+        let cursor = AnsiCursor::new();
+
+        TerminalCursor { cursor }
+    }
+
+    /// Moves the cursor to the given position.
+    pub fn goto(&self, column: u16, row: u16) -> Result<()> {
+        self.cursor.goto(column, row)
+    }
+
+    /// Gets the cursor position (`(column, row)` tuple).
+    pub fn pos(&self) -> Result<(u16, u16)> {
+        self.cursor.pos()
+    }
+
+    /// Moves the cursor `row_count` times up.
+    pub fn move_up(&mut self, row_count: u16) -> Result<&mut TerminalCursor> {
+        self.cursor.move_up(row_count)?;
+        Ok(self)
+    }
+
+    /// Moves the cursor `col_count` times right.
+    pub fn move_right(&mut self, col_count: u16) -> Result<&mut TerminalCursor> {
+        self.cursor.move_right(col_count)?;
+        Ok(self)
+    }
+
+    /// Moves the cursor `row_count` times down.
+    pub fn move_down(&mut self, row_count: u16) -> Result<&mut TerminalCursor> {
+        self.cursor.move_down(row_count)?;
+        Ok(self)
+    }
+
+    /// Moves the cursor `col_count` times left.
+    pub fn move_left(&mut self, col_count: u16) -> Result<&mut TerminalCursor> {
+        self.cursor.move_left(col_count)?;
+        Ok(self)
+    }
+
+    /// Saves the cursor position.
+    ///
+    /// See the [restore_position](struct.TerminalCursor.html#method.restore_position) method.
+    ///
+    /// # Notes
+    ///
+    /// The cursor position is stored globally and is not related to the current / any
+    /// `TerminalCursor` instance.
+    pub fn save_position(&self) -> Result<()> {
+        self.cursor.save_position()
+    }
+
+    /// Restores the saved cursor position.
+    ///
+    /// See the [save_position](struct.TerminalCursor.html#method.save_position) method.
+    pub fn restore_position(&self) -> Result<()> {
+        self.cursor.restore_position()
+    }
+
+    /// Hides the cursor.
+    ///
+    /// See the [show](struct.TerminalCursor.html#method.show) method.
+    pub fn hide(&self) -> Result<()> {
+        self.cursor.hide()
+    }
+
+    /// Shows the cursor.
+    ///
+    /// See the [hide](struct.TerminalCursor.html#method.hide) method.
+    pub fn show(&self) -> Result<()> {
+        self.cursor.show()
+    }
+
+    /// Enables or disables the cursor blinking.
+    ///
+    /// # Notes
+    ///
+    /// Windows versions lower than Windows 10 do not support this functionality.
+    pub fn blink(&self, blink: bool) -> Result<()> {
+        self.cursor.blink(blink)
+    }
+}
+
+/// Creates a new terminal cursor.
+pub fn cursor() -> TerminalCursor {
+    TerminalCursor::new()
+}
+
+/// A command to move the cursor to the given position.
+///
+/// # Notes
+///
+/// Commands must be executed / queued for execution otherwise they do nothing.
```

Comment:
Would it be an idea to link to the book in those examples?

Reply:
IIRC we agreed that the book documentation will be moved to the respective crates, mainly `lib.rs` and then the book will be completely removed until someone will have a time to write & maintain new book. Am I wrong?

Reply:
You are correct. Don't mind this comment.